viewer.py

```diff
@@ -442,7 +442,14 @@ def main():
     if settings['show_splash']:
         url = 'http://{0}:{1}/splash-page'.format(LISTEN, PORT)
 
-        get_supervisor_api_response(delay=1)
+        if is_balena_app():
+            retries = 5
+            for _ in range(retries):
+                try:
+                    get_supervisor_api_response()
+                    break
+                except requests.exceptions.ConnectionError:
+                    sleep(1)
```

Comment:
@vpetersson, after powering off and then powering on the Pi 4 (not to be confused with rebooting it) five times, the Screenly splash page consistently shows up.

Given the current changes in this commit, the only instance where **_Bad Gateway_** occurs is when I deploy new changes on my device (via `balena push [fleet-name]`).

I'll change this block to use `retry` and test the changes a few times.

If the Splash page continues to show up consistently when powered on, #1648 can be closed. A separate issue can be created for instances when the Splash page does not show up in other circumstances (e.g., during reboot, container restart only, etc.). Asking a feedback for this one...